homeassistant/components/prometheus/__init__.py

```diff
@@ -684,6 +693,63 @@ def _handle_switch(self, state: State) -> None:
 
         self._handle_attributes(state)
 
+    def _handle_fan(self, state: State) -> None:
+        metric = self._metric(
+            "fan_state", prometheus_client.Gauge, "State of the fan (0/1)"
+        )
+
+        try:
+            value = self.state_as_number(state)
+            metric.labels(**self._labels(state)).set(value)
+        except ValueError:
+            pass
+
+        fan_speed_percent = state.attributes.get(ATTR_PERCENTAGE)
+        if fan_speed_percent is not None:
+            fan_speed_metric = self._metric(
+                "fan_speed_percent",
+                prometheus_client.Gauge,
+                "Fan speed percent (0-100)",
+            )
+            fan_speed_metric.labels(**self._labels(state)).set(float(fan_speed_percent))
+
+        fan_is_oscillating = state.attributes.get(ATTR_OSCILLATING)
+        if fan_is_oscillating is not None:
+            fan_oscillating_metric = self._metric(
+                "fan_is_oscillating",
+                prometheus_client.Gauge,
+                "Fan oscillating attribute (0/1)",
+            )
+            fan_oscillating_metric.labels(**self._labels(state)).set(
+                float(fan_is_oscillating)
+            )
+
+        fan_preset_mode = state.attributes.get(ATTR_PRESET_MODE)
+        available_modes = state.attributes.get(ATTR_PRESET_MODES)
+        if fan_preset_mode and available_modes:
+            fan_preset_metric = self._metric(
+                "fan_preset_mode",
+                prometheus_client.Gauge,
+                "Fan preset mode enum",
+                ["mode"],
+            )
+            for mode in available_modes:
+                fan_preset_metric.labels(**dict(self._labels(state), mode=mode)).set(
+                    float(mode == fan_preset_mode)
+                )
+
+        fan_direction = state.attributes.get(ATTR_DIRECTION)
+        if fan_direction is not None:
+            fan_direction_metric = self._metric(
+                "fan_direction",
+                prometheus_client.Gauge,
+                "Fan direction (0=forward, 1=reverse)",
```

Comment:
I think this works, but another alternative might be to have the metric name better reflect the "positive" value of 1, e.g. `fan_direction_reversed`. Up to you.